riak_test/src/cs_suites.erl

```diff
@@ -0,0 +1,574 @@
+    lager:debug("storage stats for user ~s , ~s/~s",
+                [UserConfig#aws_config.access_key_id, Begin, End]),
+    Expected = lists:sort(
+                 [{B, {Count, bucket_bytes(B, Count)}} ||
+                     #bucket{name=B, count=Count} <- Buckets]),
+    Actual = rtcs_admin:storage_stats_json_request(AdminConfig, UserConfig,
+                                                   Begin, End),
+    ?assertEqual(Expected, Actual),
+    Circle.
+
+first_n_bytes(Binary, ByteSize) ->
+    binary:part(Binary, 0, math:max(byte_size(Binary), ByteSize)).
+
+-spec bk_to_body(string(), integer()) -> binary() | [binary()].
+bk_to_body(B, K) ->
+    case os:getenv("CS_RT_DEBUG") of
+        "true" ->
+            %% Trick to make duration/stacktrace smaller for debugging (/ω・＼)
+            bk_to_body_debug(B, K);
+        _ ->
+            %% This branch should be used normally
+            bk_to_body_actual(B, K)
+    end.
+
+%% Generate object body by modulo 5 of `K::integer()':
+%% 1    : multiple blocks  (3 MB)
+%% 2    : Multipart Upload (2 parts * 5 MB/part)
+%% 3    : Put Copy of `1'  (3 MB)
+%% 4, 0 : small objects    (~ 10 KB)
+-spec bk_to_body_actual(string(), integer()) ->
+                               binary() | [binary()] | {copy, {string(), integer()}}.
+bk_to_body_actual(_B, K) when (K rem 5) =:= 1 -> binary:copy(<<"A">>, mb(3));
+bk_to_body_actual(_B, K) when (K rem 5) =:= 2 -> lists:duplicate(2, binary:copy(<<"A">>, mb(5)));
+bk_to_body_actual(B, K)  when (K rem 5) =:= 3 -> {copy, {B, 1}};
+bk_to_body_actual(B, K)                       -> binary:copy(lead(B, K), 1024).
+
+-spec bk_to_body_debug(string(), integer()) -> binary() | [binary()].
+bk_to_body_debug(B, K) ->
+    lead(B, K).
+
+lead(B, K) ->
+    to_bin([B, $:, to_str(K), $\n]).
+
+%% Calculate total byte size for the bucket
+-spec bucket_bytes(string(), non_neg_integer()) -> non_neg_integer().
+bucket_bytes(B, Count) ->
+    lists:sum([obj_bytes(B, K) || K <- lists:seq(1, Count)]).
+
+%% Returns byte size of the binary that will be returned by
+%% `bk_to_body(B, K)'.
+-spec obj_bytes(string(), integer()) -> non_neg_integer().
+obj_bytes(B, K) ->
+    case os:getenv("CS_RT_DEBUG") of
+        "true" ->
+            %% Use OS env var to restrict object body to small binary
+            %% for debugging. no multi-MB obj, no Multipart, no PUT Copy.
+            byte_size(bk_to_body_debug(B, K));
+        _ ->
+            obj_bytes_actual(B, K)
+    end.
+
+obj_bytes_actual(_B, K) when (K rem 5) =:= 1 -> mb(3);
+obj_bytes_actual(_B, K) when (K rem 5) =:= 2 -> 2 * mb(5);
+obj_bytes_actual(B, K)  when (K rem 5) =:= 3 -> obj_bytes_actual(B, 1);
+obj_bytes_actual(B, K)                       -> kb(byte_size(lead(B, K))).
+
+kb(KB) -> KB * 1024.
+
+mb(MB) -> MB * 1024 * 1024.
+
+merge_all_bitcask(#state{riak_nodes=Nodes, riak_vsn=Vsn} = _State) ->
+    [trigger_bitcask_merge(N, Vsn) || N <- Nodes],
+    %% Wait for merge worker finished
+    [rt:wait_until(
```

Comment:
You have to verify the result of `rt_wait:until/2` before going next. In my classic old **HDD** env this got timed out and extending timeout made it success. I'd also recommend setting special retry delay here, like twice bigger than existing value, by using `rt_wait:until/3`.